File: leetcode/go/1497.check-if-array-pairs-are-divisible-by-k.go
func canArrange(arr []int, k int) bool {
    rems := make([]int, k)

    for _, n := range arr {
        rems[((n % k) + k) % k]++
    }

    if rems[0] & 1 != 0 {
        return false
    }

    for i:=1; i < 1 + k/2; i++ {
        if rems[i] != rems[k-i] {
            return false
        }
    }

    return true
}

File: leetcode/rust-leetcode/src/1497.check-if-array-pairs-are-divisible-by-k.rs
impl Solution {
    pub fn can_arrange(arr: Vec<i32>, k: i32) -> bool {
        let mut rems = vec![0; k as usize];
        for num in arr {
            rems[(((num % k) + k) % k) as usize] += 1;
        }
        if rems[0] % 2 != 0 {
            return false;
        }
        let k = k as usize;
        // println!("{:?}", rems);
        for n in 1..=k/2 {
            if rems[n] != rems[k-n] {
                return false;
            }
        }
        true
    }
}

